Allow to be autoreconfed w/o libgtk1.2-devel installed.

--- origsrc/imlib-1.9.15/configure.in	2010-02-12 13:26:00.181887300 -0600
+++ src/imlib-1.9.15/configure.in	2010-02-21 23:42:15.532180500 -0600
@@ -27,7 +27,7 @@ AC_ARG_ENABLE(gdk, [  --enable-gdk      
 if test x$disable_gdk = xno; then
  AC_MSG_RESULT(no)
 
-AM_PATH_GTK(1.2.1,[
+PKG_CHECK_MODULES(GTK, [gtk+ >= 1.2.1],[
   GDK_IMLIB="gdk_imlib utils"],[
   GDK_IMLIB="" 
   AC_MSG_WARN([*** gdk_imlib will not be built ***])])
