--- origsrc/imlib-1.9.15/Imlib/Makefile.am	2004-09-20 19:22:59.000000000 -0500
+++ src/imlib-1.9.15/Imlib/Makefile.am	2007-12-16 13:15:55.890625000 -0600
@@ -30,10 +30,10 @@
 	save.c		\
 	snprintf.c
 
-libImlib_la_LDFLAGS    = -version-info 10:15:9
+libImlib_la_LDFLAGS    = -version-info 10:15:9 -no-undefined
 #libImlib_la_LDDADD     = $(SUPPORT_LIBS)
 libImlib_la_LIBADD     = $(SUPPORT_LIBS) $(X_PRE_LIBS) \
-			 $(X_LIBS) $(X_EXTRA_LIBS)
+			 $(X_LIBS) $(X_EXTRA_LIBS) -lX11
 
 DEFS =  -DSYSTEM_IMRC=\"$(sysconfdir)/imrc\" \
 	-DSYSCONFDIR=\"$(sysconfdir)\"
--- origsrc/imlib-1.9.15/configure.in	2007-12-16 13:15:51.812500000 -0600
+++ src/imlib-1.9.15/configure.in	2007-12-16 15:33:21.171875000 -0600
@@ -102,6 +102,7 @@
 AC_SUBST(GIFLIBS)
 AC_SUBST(JPEGLIBS)
 AC_SUBST(PNGLIBS)
+AC_SUBST(XPMLIBS)
 	
 
 AC_SUBST(GDK_IMLIB)
@@ -265,12 +266,12 @@
 
 
 AC_CHECK_HEADER(gif_lib.h,
-  [AC_CHECK_LIB(ungif, DGifOpenFileName,
-    [GIFLIBS="-lungif"
-     SUPPORT_LIBS="$SUPPORT_LIBS -lungif"; AC_DEFINE(HAVE_LIBGIF, 1, [ ])],
-    [AC_CHECK_LIB(gif, DGifOpenFileName,
-      [GIFLIBS="-lgif"
-      SUPPORT_LIBS="$SUPPORT_LIBS -lgif"
+  [AC_CHECK_LIB(gif, DGifOpenFileName,
+    [GIFLIBS="-lgif"
+     SUPPORT_LIBS="$SUPPORT_LIBS -lgif"; AC_DEFINE(HAVE_LIBGIF, 1, [ ])],
+    [AC_CHECK_LIB(ungif, DGifOpenFileName,
+      [GIFLIBS="-lungif"
+      SUPPORT_LIBS="$SUPPORT_LIBS -lungif"
        AC_DEFINE(HAVE_LIBGIF, 1, [ ])],
       [AC_MSG_ERROR(*** GIF library file not found ***)], 
       $GX_LIBS)],
@@ -303,6 +304,24 @@
 	SUPPORT_LIBS="$SUPPORT_LIBS -lpng -lz"; AC_DEFINE(HAVE_LIBPNG, 1, [ ])
 fi
 
+
+AC_CHECK_LIB(jpeg, jpeg_destroy_decompress,
+  xpm_ok=yes,
+  xpm_ok=no
+  AC_MSG_ERROR(*** XPM library not found ***), $GX_LIBS)
+if test "$xpm_ok" = yes; then
+  AC_CHECK_HEADER(X11/xpm.h,
+    xpm_ok=yes,
+    xpm_ok=no)
+  AC_MSG_RESULT($xpm_ok)
+  if test "$xpm_ok" = yes; then 
+    XPMLIBS="-lXpm -lX11"
+    SUPPORT_LIBS="$SUPPORT_LIBS -lXpm"; AC_DEFINE(HAVE_LIBXPM, 1, [ ])
+  else
+    AC_MSG_ERROR(*** XPM header file not found ***)
+  fi
+fi
+
 SUPPORT_LIBS="$SUPPORT_LIBS -lm"
 GDK_SUPPORT_LIBS="$SUPPORT_LIBS"
 
@@ -319,6 +338,7 @@
 AM_CONDITIONAL(HAVE_TIFF, test "x$TIFFLIBS" != x)
 AM_CONDITIONAL(HAVE_PNG, test "x$PNGLIBS" != x)
 AM_CONDITIONAL(HAVE_JPEG, test "x$JPEGLIBS" != x)
+AM_CONDITIONAL(HAVE_XPM, test "x$XPMLIBS" != x)
 
 AC_SUBST(SUPPORT_LIBS)
 AC_SUBST(GDK_SUPPORT_LIBS)
--- origsrc/imlib-1.9.15/gdk_imlib/Makefile.am	2004-09-22 19:41:59.000000000 -0500
+++ src/imlib-1.9.15/gdk_imlib/Makefile.am	2007-12-16 13:15:55.921875000 -0600
@@ -1,4 +1,4 @@
-common_ldflags = -avoid-version -module
+common_ldflags = -avoid-version -module -no-undefined
 
 # The modules for loading/saving the various graphics formats
 # are located in $(moduledir).  Default it to libdir for backwards
@@ -16,7 +16,7 @@
 
 libimlib_png_la_SOURCES = io-png.c
 libimlib_png_la_LDFLAGS = $(common_ldflags)
-libimlib_png_la_LIBADD  = $(PNGLIBS) libgdk_imlib.la
+libimlib_png_la_LIBADD  = $(PNGLIBS) libgdk_imlib.la $(GMODULE_LIBS)
 
 libimlib_tiff_la_SOURCES = io-tiff.c
 libimlib_tiff_la_LDFLAGS = $(common_ldflags)
@@ -32,7 +32,7 @@
 
 libimlib_xpm_la_SOURCES = io-xpm.c
 libimlib_xpm_la_LDFLAGS = $(common_ldflags)
-libimlib_xpm_la_LIBADD = libgdk_imlib.la
+libimlib_xpm_la_LIBADD = $(XPMLIBS) libgdk_imlib.la
 
 libimlib_ppm_la_SOURCES = io-ppm.c
 libimlib_ppm_la_LDFLAGS = $(common_ldflags)
@@ -56,14 +56,17 @@
 TIFF = libimlib-tiff.la
 endif
 
+if HAVE_XPM
+XPM = libimlib-xpm.la
+endif
+
 OTHER = \
 	libimlib-bmp.la		\
-	libimlib-xpm.la		\
 	libimlib-ppm.la		\
 	libimlib-ps.la
 
 DYNAMIC_LIBS = \
-	$(OTHER) $(GIF) $(JPEG) $(TIFF) $(PNG)
+	$(OTHER) $(GIF) $(JPEG) $(TIFF) $(PNG) $(XPM)
 
 else
 
@@ -86,7 +89,7 @@
 	modules.c
 
 
-libgdk_imlib_la_LDFLAGS = -version-info 10:15:9
+libgdk_imlib_la_LDFLAGS = -version-info 10:15:9 -no-undefined
 libgdk_imlib_la_LIBADD = $(GX_LIBS)
 
 
--- origsrc/imlib-1.9.15/gdk_imlib/modules.c	1999-07-30 11:34:02.000000000 -0500
+++ src/imlib-1.9.15/gdk_imlib/modules.c	2007-12-16 13:15:55.937500000 -0600
@@ -83,7 +83,11 @@
 	gboolean v;
 	void *ptr;
 
+#ifdef __CYGWIN__
+	modname = g_strconcat ("cygimlib-", mod, ".dll", NULL);
+#else
 	modname = g_strconcat ("imlib-", mod, NULL);
+#endif
 	path = g_module_build_path (IMLIB_LIB, modname);
 	g_free (modname);
 
